lib/elixir/test/elixir/io/ansi_test.exs

```diff
@@ -155,4 +155,53 @@ defmodule IO.ANSITest do
       IO.ANSI.color_background(5, -1, 1)
     end
   end
+
+  test "cursor/2" do
+    assert IO.ANSI.cursor(0, 0) == "\e[0;0H"
+    assert IO.ANSI.cursor(11, 12) == "\e[11;12H"
+
+    assert_raise FunctionClauseError, fn ->
+      IO.ANSI.cursor(-1, 5)
+    end
+
+    assert_raise FunctionClauseError, fn ->
+      IO.ANSI.cursor(5, -1)
+    end
+  end
+
+  test "cursor_up/1" do
+    assert IO.ANSI.cursor_up() == "\e[1A"
+    assert IO.ANSI.cursor_up(12) == "\e[12A"
+
+    assert_raise FunctionClauseError, fn ->
```

Comment:
Should we also test the value `0`? it seems a sensitive value because in an alternative implementation could be interpreted as `\e[A` or `\e[0A`. `\e[0A` works on my terminal and it's equivalent to `\e[A` or `\e[1A`.

Reply:
Good catch! Wouldn't it be best to have a guard that the value is `>= 1`? :)

Reply:
If we go that way, the typespec should use `pos_integer()`

Reply:
I have the guard as `> 0`, but I didn't realize `pos_integer()` is a built-in. will change to that

Reply:
I had the guard as `> 0` but `>= 1` is clearer. I changed that and the typespecs